pkg/runtime/reconciler.go

```diff
@@ -268,7 +268,21 @@ func (r *resourceReconciler) Sync(
 			rlog.Info("updated resource")
 		}
 	}
-	if ackcompare.IsNotNil(latest) {
+	rlog.Enter("rm.LateInitialize")
+	lateInitializedLatest, err := rm.LateInitialize(ctx, latest)
+	rlog.Exit("rm.LateInitialize", err)
+	// LateInitialize returns *requeue.RequeueNeededAfter. For the pointer simple nil check does not work
+	// Use ackcompare.IsNotNil() method
+	if ackcompare.IsNotNil(err) {
```

Comment:
Why simple nil checks does not work with `err` here?

Reply:
Because LateInitialize return "*requeue.RequeueNeededAfter", which does implement Error() interface but falls into trap of pointer nil comparison. I did add the explanation in comments.

Reply:
Agreed, should just use simple nil check for err.